addons/mail/models/statusbar_time_mixin.py

```diff
@@ -0,0 +1,76 @@
+import logging
+from datetime import timedelta
+
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class StatusbarTimeMixin(models.AbstractModel):
+    _name = "statusbar.time.mixin"
+    _description = "Mixin to compute the time a record spent in each stage"
+    _stage_field = "stage_id"
+
+    json_status_time = fields.Json(
+        string="Status time", compute="_compute_json_status_time", help="JSON that maps stage ids to seconds spent")
+
+    def _compute_json_status_time(self):
+        """
+        Computes json_status_time, a Json field stored as { <stage_id (str)>: <time_spent_in_seconds (int)> }
+
+            e.g. {"1": 1230, "3": 2220, "5": 14}
+        """
+
+        field = self.env['ir.model.fields'].sudo().search_fetch([
+            ('model', '=', self._name),
+            ('name', '=', self._stage_field),
+        ], ['id', 'tracking'], limit=1)
+        if (field.tracking == 0):
+            self.json_status_time = False
+            _logger.warning(
+                'Field "%s" on model "%s" must have tracking=True for the computation of statusbar time.',
+                self._stage_field, self._name
+            )
+            return
+
+        query = """
+               SELECT m.res_id,
+                      v.create_date,
+                      v.old_value_integer,
+                      v.new_value_integer
+                 FROM mail_tracking_value v
+            LEFT JOIN mail_message m
+                   ON m.id = v.mail_message_id
+                  AND v.field = %(field_id)s
+                WHERE m.model = %(model_name)s
+                  AND m.res_id IN %(record_ids)s
+             ORDER BY v.id
+        """
+        self.env.cr.execute(query, {"field_id": field.id, "model_name": self._name, "record_ids": tuple(self.ids)})
```

Comment:
We could add an early return at the beginning of the method when being in create mode so that we avoid useless queries, like
```
if not self.ids:
    self.json_status_time = False
    return
...
```

Reply:
This doesn't seem necessary as when creating a record, the compute method is not triggered, unless I'm missing something?